### PR TITLE
Feature/blui 4367 header md3

### PR DESCRIPTION
**Pull request by ArshdeepSingh3-eaton**
Changes proposed in this Pull Request:
- Add Header Component
- 
- 

Screenshots / Screen Recording (if applicable)
-
Screenshot 2023-10-10 at 2 00 13 PM

Screenshot 2023-10-10 at 2 00 22 PM

To Test:
-run yarn start:showcase-ios

### CODE REVIEW

**Comment by huayunh** [comment]
Why are the Android & iOS screenshots looking different?

**Comment by ArshdeepSingh3-eaton** [comment]
Update the code and Screenshots

**Review by surajeaton — COMMENTED** [review]
Please resolve the conflicts

**Comment by huayunh** [comment]
From the screenshot, I do not see any drop shadows under the Android version. The iOS version rendered the shadow just fine. It appears that the two platforms treat shadow differently?

Also the Material Design 3 app bars look so different from Material Design 2's. I have some design thoughts on that, but that's going to require quite a few changes on the dev side, so I am willing to look away haha.

File: components/src/md3/Header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,748 @@
...
+            // @change color once have a correct color
+            // (theme.dark ? theme.colors.actionPalette?.active || theme.colors.surface : theme.colors.primary),
+            (theme.dark ? theme.colors.surface : theme.colors.primary),
+        shadowColor: 'rgba(0, 0, 0, 0.3)',
+        shadowOffset: {
+            width: 0,
+            height: 1,
+        },
+        shadowRadius: 2,
+        shadowOpacity: 1,
```
Comment (huayunh) [inline components/src/md3/Header/Header.tsx:57]: Is there any chance that we can include shadow styles in a theme? If so, can you write a story/modify an existing story to do it?

Comment (ArshdeepSingh3-eaton) [reply components/src/md3/Header/Header.tsx:57]: I think that's not possible to include shadow in theme as no properties are present
the theme consists on mainly FONT and COLOUR

**Review by surajeaton — COMMENTED** [review]
I am getting error on iOS. In showcase, I have checked out feature/blui-4367-header-md3 this branch.
Simulator Screenshot - iphone 12 pro - 2023-10-11 at 15 39 21

**Comment by ektaghag-eaton** [comment]
@huayunh , @ArshdeepSingh3-eaton where I can find the design here https://www.figma.com/file/UOzL2R0dlgxozVKE4A4yRV/Brightlayer-UI-Tokens-%2B-Components-(in-Material-Design-3)-(Community)?type=design&node-id=19-261&mode=design&t=EJF0f4U9jqtV15hA-0 ?

**Review by ektaghag-eaton — COMMENTED** [review]
1. Please check design specifications for header here - https://brightlayer-ui.github.io/patterns/appbar#design-specifications

2. Title and subtitle font size looks correct
3. FontSize for Info is incorrect
4. In header sequence should be `Title, Subtitle and then Info`, but currenlty it is `Title, Info and then Subtitle`. Need to change sequence.
Screenshot 2023-10-12 at 2 48 34 PM

**Comment by ektaghag-eaton** [comment]
The build is failing , can you run `yarn precommit`?

**Comment by ArshdeepSingh3-eaton** [comment]
this pipeline issue is being resolved

already working on a story to resolve that

File: components/src/md3/Header/headerContent.tsx [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+    const getSubtitleStyle = useCallback(
+        () => ({
+            color: textColor,
+            fontFamily: 'OpenSans-Regular',
+            fontSize: headerHeight.interpolate({
+                inputRange: [REGULAR_HEIGHT, EXTENDED_HEIGHT],
+                outputRange: [14, 16],
+                extrapolate: 'clamp',
+            }),
+            // fontFamily: theme.fonts.titleMedium,
```
Comment (ektaghag-eaton) [inline components/src/md3/Header/headerContent.tsx:131]: Remove commented code.

**Review by ektaghag-eaton — APPROVED** [review]

**Pull request merged**